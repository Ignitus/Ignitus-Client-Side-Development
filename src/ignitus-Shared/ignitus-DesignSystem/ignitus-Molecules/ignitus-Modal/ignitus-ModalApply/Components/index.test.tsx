import React from 'react';
import { mount, render } from 'enzyme';
import { ModalApply } from './index';

it('should render', () => {
  const tree = render(<ModalApply />);
  expect(tree).toMatchSnapshot();
});

describe('<ModalApply />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<ModalApply />);
  });

  it('should render input, span, button and p tags', () => {
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('input').length).toBe(1);
    expect(wrapper.find('button').length).toBe(3);
    expect(wrapper.find('p').length).toBe(3);
  });

  it('should render default p text', () => {
    expect(
      wrapper
        .find('p')
        .at(0)
        .text(),
    ).toBe('Email *');
    expect(
      wrapper
        .find('p')
        .at(1)
        .text(),
    ).toBe('Resume *');
    expect(
      wrapper
        .find('p')
        .at(2)
        .text(),
    ).toBe('Cover letter');
  });

  it('should render default button text', () => {
    expect(
      wrapper
        .find('button')
        .at(0)
        .text(),
    ).toBe('Upload Document');
    expect(
      wrapper
        .find('button')
        .at(1)
        .text(),
    ).toBe('Apply');
    expect(
      wrapper
        .find('button')
        .at(2)
        .text(),
    ).toBe('Cancel');
  });
});
