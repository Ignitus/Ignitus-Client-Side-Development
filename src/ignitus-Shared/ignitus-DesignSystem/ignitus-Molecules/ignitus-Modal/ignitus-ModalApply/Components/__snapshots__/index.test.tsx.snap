// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  class="css-xpwoj3"
>
  <div
    class="css-w72ftu"
  >
    <p
      class="css-9adi7w"
    >
      Email *
    </p>
    <input
      class="css-v8hd27"
      height="3rem"
      name="name"
      placeholder="Email"
      type="text"
      width="450px"
    />
  </div>
  <div
    class="css-3oqt7t"
  >
    <div
      class="css-w72ftu"
    >
      <p
        class="css-9adi7w"
      >
        Resume *
      </p>
      <span
        class="css-1ofj0nk"
        color="#ffffff"
      >
        Sophia_resume.pdf
      </span>
    </div>
    <div
      class="css-1bz476y"
    >
      <p
        class="css-9adi7w"
      >
        Cover letter
      </p>
      <button
        class="css-1x1p5xb"
      >
        Upload Document
      </button>
    </div>
  </div>
  <div
    class="css-m74gbk"
  >
    <button
      class="css-1w1qkgh"
    >
      Apply
    </button>
    <button
      class="css-da71r7"
    >
      Cancel
    </button>
  </div>
</div>
`;
